tests/plugins/clipboard/manual/customtypes.html

```diff
@@ -0,0 +1,84 @@
+<h2>Classic editor</h2>
+
+<div id="classic">
+	<h1>Classic editor</h1>
+
+	<p>Copy content and paste content here.</p>
+</div>
+
+<style>
+#output{ width: 500px;border: 1px solid black; table-layout: fixed; }
+#output th:first-child { width: 70px; }
+#output tbody td{ border-top: 1px solid black; white-space: pre-wrap; }
+#output tbody tr:nth-child(odd) { background: #F3F3F3; }
+#output tbody tr.raw { background: rgb(242, 243, 194); }
+</style>
+
+<table id="output">
+	<thead>
+		<tr>
+			<th>Type</th>
+			<th>Value</th>
+		</tr>
+	</thead>
+	<tbody></tbody>
+	<caption>Pasted data transfer dump</caption>
+</table>
+
+<script>
+	CKEDITOR.replace( 'classic', {
+		on: {
+			instanceReady: function( evt ) {
+				evt.editor.editable().on( 'copy', function( ckEvt ) {
+					var evt = ckEvt.data.$,
+						nativeDataTransfer = evt.clipboardData,
+						ckDataTransfer = new CKEDITOR.plugins.clipboard.dataTransfer( nativeDataTransfer );
+
+					ckDataTransfer.setData( 'text/plain', 'plain value' );
+					ckDataTransfer.setData( 'text/html', 'html value' );
+					ckDataTransfer.setData( 'foo-bar', 'foo-bar value' );
+					ckDataTransfer.setData( 'baz', 'baz value' );
+
+					// Prevent DOM event, so that clipboard gets overriden.
+					evt.preventDefault();
+				}, null, null, 100000 );
+
+				evt.editor.editable().on( 'paste', function( evt ) {
+					// This is native approach.
+					var tbody = CKEDITOR.document.findOne( 'tbody' ),
+						nativeDataTransfer = evt.data.$.clipboardData,
+						ckDataTransfer = new CKEDITOR.plugins.clipboard.dataTransfer( nativeDataTransfer ),
+						output = '',
+						types;
+
+					// Force data cache so that we can acces data types from cache.
+					ckDataTransfer.cacheData();
+
+					// Eventually these types should be dynamically fetched.
+					types = [ 'Text', 'text/html', 'foo-bar', 'baz' ];
+
+					function getRow( row1, row2, rowClass ) {
+						return '<tr class="' + rowClass + '"><td>' + row1 + '</td><td>' + CKEDITOR.tools.htmlEncode( row2 ) + '</td></tr>';
+					}
+
+					for ( var j = 0; j < types.length; j++ ) {
+						output += getRow( types[ j ], ckDataTransfer.getData( types[ j ] ) );
+					}
+
+					// Extract raw HTML, see
+					if ( CKEDITOR.tools.array.indexOf( types, 'text/html' ) != -1 ) {
+						// Native 'getData' needs to be used here as 'ckDataTransfer.getData'
+						// strips special comment which we would like to show here too.
+						output += getRow( 'Native HTML data', ckDataTransfer.$.getData( 'text/html', true ), 'raw' );
```

Comment:
Makes sense 👍 , I wasn't fully sure how it should work with fallback html.

Reply:
Btw. should `getNative` work only for `text/html`? In Edge if you call `getData( 'cke/id', true )` if we assume `getNative` should really return what natively sits in dataTransfer you will get empty string.

Reply:
The other issue with `getNative` flag is that, it is used inside [`cacheData`](https://github.com/ckeditor/ckeditor-dev/blob/96bf4037cb3ff812fc9b993bce4a304a884f361a/plugins/clipboard/plugin.js#L2439):
https://github.com/ckeditor/ckeditor-dev/blob/96bf4037cb3ff812fc9b993bce4a304a884f361a/plugins/clipboard/plugin.js#L2450

so we would like to cache "native value" for all browsers, but for Edge the cache should contain proper `type` : `value` pairs without special comment so it shouldn't basically use `getNative` (`fallbackDataTransfer.isRequired()` could be use here but it will create another "logic branch", increasing the complexity of already complex plugin 😞 ).

Reply:
> Btw. should getNative work only for text/html? In Edge if you call getData( 'cke/id', true ) if we assume getNative should really return what natively sits in dataTransfer you will get empty string.

This is exactly how it should work, I can  see that you already implemented it that way. I have corrected this in 33757361d.